File: 2022.01.15/javascript/4673.js
let visited = new Array(10001).fill(false);
let next_num = 0;
let d = function (n){
    let num = n;
    while (n){
        num += n%10;
        n = parseInt(n/10);
    }
    return num;
}
for (let i = 0 ; i<visited.length ; i++){
    next_num = d(i);
    if (next_num>10000 || visited[next_num]){
        continue;
    }
    visited[next_num] = true;
}
for (let i = 1 ; i<visited.length ; i++){
    if (!visited[i]){
        console.log(i);
    }
}

File: Boj.단계별/6_함수/python/4673.py
visited = [False for _ in range(10001)]
def d(n):
    num = n
    while n:
        num += n%10
        n //= 10
    return num
for i in range(10001):
    num = d(i)
    if num<=10000 and not visited[num]:
        visited[num] = True
for i in range(len(visited)):
    if not visited[i]:
        print(i)